runtime/src/iree/hal/device.h

```diff
@@ -467,6 +468,10 @@ typedef struct iree_hal_device_vtable_t {
                                          iree_string_view_t key,
                                          int64_t* out_value);
 
+  iree_status_t(IREE_API_PTR* create_channel)(
```

Comment:
Can you think a bit about how to test this HAL code?

The HAL CTS allows for including and excluding tests: https://github.com/iree-org/iree/blob/main/runtime/src/iree/hal/drivers/cuda/cts/CMakeLists.txt so you could add a test that only runs on CUDA at the start, for example.

Reply:
CTS should may work for this stuff but if we need multiple processes collaborating it's likely to need its own dedicated runner to set that up. That could still be a C test that forks off other processes, though.